[master<873] Backup and restore update (#876)
memgraph/docs: docs/how-to-guides/create-backup.md

title: How to backup and restore data
sidebar_label: Backup and restore data
---
import Tabs from "@theme/Tabs"; import TabItem from "@theme/TabItem";

While running, Memgraph generates various files in its [data
directory](/docs/memgraph/reference-guide/backup), including the **durability
files**, that is,  snapshots and WALs that contain Memgraph's data in a
recoverable format and are located in the `wal` and `snapshots` folders in the

> data directory. On startup, Memgraph searches for previously saved durability

Author: i know, but... 
1. this is a how to so this is irrelevant to this how to
2. i don't want to list them ALL because then we will have two places to update if something changes... MVP

  If you've just created a snapshot file there is no need to backup WAL files. 

  To help copying the files from the Docker container, check out the [Working with
  docker
  guide](/how-to-guides/work-with-docker.md#how-to-copy-files-from-and-to-a-docker-container).

4. **Unlock the data directory**

  Run the following query in `mgconsole` or Memgraph Lab to unlock the
  directory:

  ```cypher
  UNLOCK DATA DIRECTORY;
  ```

  Memgraph will delete the files which should have been deleted before locking and
  allow any future deletion of the durability files.

## Restore data

To restore data from a backup

<Tabs
  groupId="backup"
  defaultValue="docker"
  values={[
    {label: 'Docker 🐳', value: 'docker'},
    {label: 'Linux', value: 'linux'}
  ]}>
<TabItem value='docker'>

1. Empty the `wal` directory

  If you want to restore data only from the snapshot file, ensure that the
  `wal` directory is empty:

     - Find the container ID using a `docker ps` command, then enter the container using:

    ```
    docker exec -it CONTAINER_ID bash
    ```
     - Position yourself in the `/var/lib/memgraph/wal` directory and `rm *`

> 1. Stop the instance using `docker stop CONTAINER_ID`

Reviewer: Maybe mention the potential creation of new snapshot on exit.

> 1. Stop the instance using `docker stop CONTAINER_ID`

Author: Not sure why it is relevant when you are c/p the whole directory with the one you have on disk... (this is a how-to, not a reference guide so only info relevant to this process are written)
So whatever i have in the container will be replaced with my back-up dir

2. Start the instance by adding a `-v ~/snapshots:/var/lib/memgraph/snapshots`
  flag to the `docker run` command, where the `~/snapshots` represents a path

>   to the location of the backed-up directory, for example: 

Author: ```suggestion
  to the location of the directory with the back-up snapshot, for example: 
```

  ```

>   docker run -p 7687:7687 -p 7444:7444 -v ~/snapshots:/var/lib/memgraph/snapshots memgraph/memgraph

Author: ```suggestion
  docker run -p 7687:7687 -p 7444:7444 -v C:/Users/Vlasta/snapshots:/var/lib/memgraph/snapshots memgraph/memgraph
```

  ```
4. If you want to copy both WAL and snapshot files start the instance by adding
  a `-v ~/snapshots:/var/lib/memgraph/snapshots -v ~/wal:/var/lib/memgraph/wal`
  flags to the `docker run` command, where the `~/snapshots` represents a path
  to the location of the backed-up snapshot directory, and `~/wal` represents a
  path to the location of the backed-up wal directory for example: 

  ```

>   docker run -p 7687:7687 -p 7444:7444 -v ~/snapshots:/var/lib/memgraph/snapshots -v ~/wal:/var/lib/memgraph/wal memgraph/memgraph

Author: ```suggestion
  docker run -p 7687:7687 -p 7444:7444 -v C:\Users\Vlasta\snapshots:/var/lib/memgraph/snapshots -v C:/Users/Vlasta/wal:/var/lib/memgraph/wal memgraph/memgraph
```

  ```

</TabItem>
<TabItem value='linux'>

1. Before running an instance, copy the backed up snapshot into the `snapshots`

>    directory, and optionally, copy the backed-up wal files into the `wal`

Author: ```suggestion
   directory, and optionally, copy the backed-up WAL files into the `wal`
```
